torch/utils/data/dataframes_pipes.ipynb

```diff
@@ -0,0 +1,437 @@
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "DataFrames Pipe\n   i  j\n0  0  0\n   i  j\n1  1  1\n   i  j\n2  2  2\n   i  j\n3  3  0\n   i  j\n4  4  1\n   i  j\n5  5  2\n   i  j\n6  6  0\n   i  j\n0  7  1\n   i  j\n1  8  2\n   i  j\n2  9  0\nRegular DataPipe\n(0, 0)\n(1, 1)\n(2, 2)\n(3, 0)\n(4, 1)\n(5, 2)\n(6, 0)\n(7, 1)\n(8, 2)\n(9, 0)\n"
+     ]
+    }
+   ],
+   "source": [
+    "print('DataFrames Pipe')\n",
+    "dp = get_dataframes_pipe()\n",
+    "for i in dp:\n",
+    "    print(i)\n",
+    "\n",
+    "print('Regular DataPipe')\n",
+    "dp = get_regular_pipe()\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "You can iterate over raw DF using `raw_iterator`"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "   i  j\n0  0  0\n1  1  1\n2  2  2\n3  3  0\n4  4  1\n5  5  2\n6  6  0\n   i  j\n0  7  1\n1  8  2\n2  9  0\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = get_dataframes_pipe()\n",
+    "for i in dp.raw_iterator():\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "Operations over DF Pipe is captured"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "var_4 = input_var_3.i * 100\nvar_5 = var_4 + input_var_3.j\nvar_6 = var_5 - 2.7\ninput_var_3[\"y\"] = var_6\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = get_dataframes_pipe(dataframe_size = 3)\n",
+    "dp['y'] = dp.i * 100 + dp.j - 2.7\n",
+    "print(dp.ops_str())\n"
+   ]
+  },
+  {
+   "source": [
+    "Captured operations executed on `__next__` calls of constructed DataPipe"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "   i  j      y\n0  0  0   -2.7\n1  1  1   98.3\n2  2  2  199.3\n   i  j      y\n0  3  0  297.3\n1  4  1  398.3\n2  5  2  499.3\n   i  j      y\n0  6  0  597.3\n1  7  1  698.3\n2  8  2  799.3\n   i  j      y\n0  9  0  897.3\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = get_dataframes_pipe(dataframe_size = 3)\n",
+    "dp['y'] = dp.i * 100 + dp.j - 2.7\n",
+    "for i in dp.raw_iterator():\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "`shuffle` of DataFramePipe effects rows in individual manner"
```

Comment:
Does that mean `shuffle` is not local to each DataFrame? (from the code looks so :) )